autodiff-template

```diff
@@ -176,73 +177,80 @@ GradNum2 div(in GradNum2 a, in GradNum2 b)
 	GradNum2 r = GradNum2(a.val / b.val, a.g);
 	r.g = r.g - r.val*b.g;
 	r.g = r.g / b.val;
-	
 	return r;
-	
 }
+
 GradNum2 div(in GradNum2 a, float b)
 {
 	return GradNum2(a.val/b, a.g/b);
 }
 
+GradNum2 div(float a, in GradNum2 b)
```

Comment:
This function doesn't add a whole lot, since the caller could just do `div(a_constG2(a), b)`.

I implemented it for consistency. I find it useful to be able to call `div(a, b)` and not have to care about whether `a` and `b` are floats or autodiff types.

I defer to you though - let me know if you don't like this!